builder/buffer.go

```diff
@@ -103,30 +106,43 @@ func (b *Buffer) WritePlaceholder() {
 	}
 }
 
+// WriteField writes table and field name.
+func (b *Buffer) WriteField(table, field string) {
+	b.WriteString(b.escape(table, field))
+}
+
 // WriteEscape string.
 func (b *Buffer) WriteEscape(value string) {
-	b.WriteString(b.escape(value))
+	b.WriteString(b.escape("", value))
 }
 
-func (b Buffer) escape(value string) string {
+func (b Buffer) escape(table, value string) string {
 	if value == "*" {
-		return value
+		if table == "" {
+			return value
+		}
+		return b.Quoter.ID(table) + ".*"
 	}
 
-	key := escapeCacheKey{value: value, quoter: b.Quoter}
+	key := escapeCacheKey{table: table, value: value, quoter: b.Quoter}
 	escapedValue, ok := escapeCache.Load(key)
 	if ok {
 		return escapedValue.(string)
 	}
 
 	if len(value) > 0 && value[0] == UnescapeCharacter {
 		escapedValue = value[1:]
+	} else if regexNumber.MatchString(value) {
```

Comment:
This adds option to order/group by column number

Reply:
will it improve performance if we change this to: `value[0] >= '0' && value[0] <= '9' && regexNumber.MatchString(value)`? 🤔